### DISCUSSION TITLE
Is it possible with liquidsoap to get the samplerate, bitrate and codec of a stream ?

### DISCUSSION THREAD

**Discussion by EliXorZz** [savonet/liquidsoap#3976] [Q&A] [answered]
Hello,

I think it's possible to recover the samplerate, bitrate and codec of an input stream.
I've looked in the documentation and I've only found one way to get the samplerate [here](https://www.liquidsoap.info/doc-1.4.4/reference.html#audio.samplerate), but I don't really know how to use it and if it corresponds to what I want.

Thanks in advance

**Comment by vitoyucepi** [comment, marked as answer]
Hi @EliXorZz,
These functions probably don't exist in the liquidsoap language yet. If you really want them as part of the language, please write a feature request.
Alternatively, you could use an external tool like ffprobe to extract the necessary data.

**Comment by toots** [comment]
Hi all. @vitoyucepi is correct, we don't have inspection capacity as of now. It could be possible using the ffmpeg binding but, in general, you can also use `ffprobe` which is better designed for it. It has a `json` output that is machine-readable too, we use it in our internal testing.
